<?xml version="1.0"?>
<launch>

    <node name="map_server" pkg="map_server" type="map_server" args="$(find navigation)/map/rmuc_2025.yaml" output="screen">
        <param name="frame_id" value="map" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 -1.5708 0 0 map odom 0.1" />
    <node pkg="tf" type="static_transform_publisher" name="odom2camera_init" args="0 0 0 0 0 0 odom camera_init 0.1" />
    <node pkg="tf" type="static_transform_publisher" name="baselink2body" args="0 0 0 0 0 0 body base_link 0.1" />
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigation)/config/rivz/simu.rviz"/>
  
  <!--Launch the move base with time elastic band-->
    <param name="/use_sim_time" value="true"/>
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation)/config/global_costmap_params.yaml" command="load" />

        <param name="base_global_planner" value="global_planner/GlobalPlanner" />
        <param name="neutral_cost" value="220" />
        <param name="planner_frequency" value="3.0" />
        <param name="planner_patience" value="5.0" />
        <param name="use_dijkstra" value="false" />
        
        <!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
        <param name="controller_frequency" value="20.0" />
        <param name="controller_patience" value="15.0" /> -->

        <param name="clearing_rotation_allowed" value="false" />
        <!-- <remap from="/cmd_vel" to="/" /> -->
    </node>

  <!-- <include file="$(find auto_nav)/launch/pid_follow_planner.launch"/> -->

</launch>